File: src/main/java/toker/warbandscripts/panel/entity/Chest.java
package toker.warbandscripts.panel.entity;

import toker.warbandscripts.panel.entity.pk.ChestPK;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Collection;
import java.util.List;
import java.util.Objects;

@Entity
@Table(name = "chest")
@IdClass(ChestPK.class)
public class Chest {

    private Integer index;
    private Server server;
    private String name;
    private Integer size;

    private Collection<ChestSlot> slots;

    @Id
    @Column(name = "index", nullable = false)
    public Integer getIndex() {
        return index;
    }

    public void setIndex(Integer index) {
        this.index = index;
    }

    @Id
    @ManyToOne
    @JoinColumn(name = "server_id", referencedColumnName = "id", nullable = false)
    public Server getServer() {
        return server;
    }

    public void setServer(Server server) {
        this.server = server;
    }

    @Column(name = "name", nullable = false, length = 64)
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Column(name = "size", nullable = false)
    public Integer getSize() {
        return size;
    }

    public void setSize(Integer size) {
        this.size = size;
    }

    @OneToMany(mappedBy = "chest")
    public Collection<ChestSlot> getSlots() {
        return slots;
    }

    public void setSlots(Collection<ChestSlot> slots) {
        this.slots = slots;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Chest chest = (Chest) o;
        return Objects.equals(index, chest.index) &&
                Objects.equals(server.getId(), chest.server.getId());
    }

    @Override
    public int hashCode() {
        return Objects.hash(index, name, size);
    }
}

File: src/main/java/toker/panel/entity/Chest.kt
package toker.panel.entity

import toker.panel.entity.pk.ChestPK
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "chest")
@IdClass(ChestPK::class)
class Chest(
        @Id
        @Column(name = "index", nullable = false)
        var index: Int? = null,

        @Id
        @ManyToOne
        @JoinColumn(name = "server_id", referencedColumnName = "id", nullable = false)
        var server: Server? = null,

        @Column(name = "name", nullable = false, length = 64)
        var name: String? = null,

        @Column(name = "size", nullable = false)
        var size: Int? = null,
) {
    @OneToMany(mappedBy = "chest")
    var slots: MutableSet<ChestSlot>? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as Chest
        if (index != other.index) return false
        if (server != other.server) return false
        return true
    }

    override fun hashCode(): Int {
        var result = index ?: 0
        result = 31 * result + (server?.hashCode() ?: 0)
        return result
    }
}